File: uwp/IotCoreGatewayCpp/IotCoreGatewayCs/StartupTask.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Net.Http;
using Windows.ApplicationModel.Background;

// The Background Application template is documented at http://go.microsoft.com/fwlink/?LinkID=533884&clcid=0x409

namespace IotCoreGatewayCs
{
    public sealed class StartupTask : IBackgroundTask
    {
        BackgroundTaskDeferral deferral;
        Microsoft.Azure.IoT.Gateway.Gateway gateway;
        public void Run(IBackgroundTaskInstance taskInstance)
        {
            deferral = taskInstance.GetDeferral();

            var modules = new List<Microsoft.Azure.IoT.Gateway.IGatewayModule>();
            modules.Add(new SetOfCsModules.Module2());
            modules.Add(new SetOfCsModules.Module1());
            modules.Add(new SetOfCsModules.Module3());

            gateway = new Microsoft.Azure.IoT.Gateway.Gateway(modules);
        }
    }
}

File: uwp/IotCoreGatewayCpp/IotCoreGatewayCpp/StartupTask.cpp
#include "pch.h"
#include "StartupTask.h"

using namespace IotCoreGatewayCpp;

using namespace Platform;
using namespace Platform::Collections;
using namespace Windows::ApplicationModel::Background;
using namespace Microsoft::Azure::IoT::Gateway;
using namespace SetOfCppModules;

// The Background Application template is documented at http://go.microsoft.com/fwlink/?LinkID=533884&clcid=0x409

void StartupTask::Run(IBackgroundTaskInstance^ taskInstance)
{
	deferral = taskInstance->GetDeferral();

	auto modules = ref new Vector<IGatewayModule^>();
	modules->Append(ref new Module2());
	modules->Append(ref new Module1());
	modules->Append(ref new Module3());
	
	gateway = ref new Gateway(modules);
}

